connection-api/src/main/java/org/terracotta/connection/DiagnosticsConfig.java

```diff
@@ -0,0 +1,42 @@
+/*
+ *
+ *  The contents of this file are subject to the Terracotta Public License Version
+ *  2.0 (the "License"); You may not use this file except in compliance with the
+ *  License. You may obtain a copy of the License at
+ *
+ *  http://terracotta.org/legal/terracotta-public-license.
+ *
+ *  Software distributed under the License is distributed on an "AS IS" basis,
+ *  WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for
+ *  the specific language governing rights and limitations under the License.
+ *
+ *  The Covered Software is Terracotta Core.
+ *
+ *  The Initial Developer of the Covered Software is
+ *  Terracotta, Inc., a Software AG company
+ *
+ */
+package org.terracotta.connection;
+
+import java.util.Properties;
+
+/**
+ *
+ */
+public class DiagnosticsConfig {
+  private final Properties properties;
+  private final Runnable close;
+
+  public DiagnosticsConfig(Properties properties, Runnable close) {
+    this.properties = properties;
+    this.close = close;
+  }
+
+  public Properties getProperties() {
+    return properties;
+  }
+
+  public Runnable getClose() {
```

Comment:
How this works ? What is responsible to call `getClose().run()` to close the connection created by `DiagnosticsFactory` ?

Reply:
This has to be here for backwards compatibility and not created directly by the user.  The DiagnosticsFactory.connect is a convenience factory to create diagnostic connections.

Reply:
Ok so if I understand, no client should use `DiagnosticsFactory` because if so, it would have no way to properly close the created connection right ? These classes are just there for some reasons but we shouldn't use them ?

Reply:
Here is the usage pattern 
```
try (Diagnostics d = DiagnosticsFactory.create(<inetaddr>, props)) {
   System.out.println(d.threadDump());
}
```
versus having to create the connection and pull the entity reference.  When the Diagnostics entity is closed the underlying connection will be closed as well.